### DISCUSSION TITLE
Notification Permission Request Not Triggering Dialog

### DISCUSSION THREAD

**Discussion by khaledmurtaja** [gdelataillade/alarm#288] [Q&A]
Hi,

I noticed in the checkNotificationPermission function that it requests notification permission if the status is denied. However, while debugging the example app, no dialog appeared when this function was executed.

Note: I manually turned off the notification permission for the app before testing.
 
**This code is from example app**

`static Future<void> checkNotificationPermission() async {
  final status = await Permission.notification.status;
  if (status.isDenied) {
    alarmPrint('Requesting notification permission...');
    final res = await Permission.notification.request();
    alarmPrint(
      'Notification permission ${res.isGranted ? '' : 'not '}granted',
    );
  }
}
`

**Comment by gdelataillade** [comment]
I believe if no dialog appears then it means permission is granted by default, or that it was previously denied. Let me know if you have any questions.
